### PR TITLE
change deprecated methods

### PR DESCRIPTION
**Pull request by cewi**
## Description
changes some deprecated methods
minimum cakephp version is now 3.5 (not sure if necessary)

### CODE REVIEW

**Comment by cewi** [comment]
ok, but develop is for bootstrap4 and master for bootstrap3. So there will be seperate branches in future? Looking forward -,)

**Comment by ADmad** [comment]
@cewi Please reopen the PR :) Master won't be merged into develop, due to difference in bootstrap version it would create lot of conflicts.

**Comment by dereuromark** [comment]
I also said "might" - and "some", not "all" :)

**Comment by cewi** [comment]
no problem. HTH

**Comment by ADmad** [comment]
@cewi You need to update the deprecations in test cases too :slightly_smiling_face:.

**Comment by cewi** [comment]
@ADmad: There a still a lot of tests against Form->input() method in FormHelperTest.php. They do not throw deprecation errors. Should they be updated anyway?

**Comment by ADmad** [comment]
@cewi Yes please.

File: tests/TestCase/View/Helper/FormHelperTest.php [hunk 1]
```diff
@@ -81,28 +91,11 @@ public function tearDown()
         I18n::setLocale($this->locale);
     }
 
-    public function testBasicTextInput()
```
[the remark this answers is not in the document]
Comment (cewi) [reply tests/TestCase/View/Helper/FormHelperTest.php]: yes exactly

File: tests/TestCase/View/Helper/FlashHelperTest.php [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 use BootstrapUI\View\Helper\FlashHelper;
 use Cake\Core\Exception\Exception;
-use Cake\Network\Request;
-use Cake\Network\Session;
+use Cake\Http\ServerRequest;
+use Cake\Http\Session;
```
Comment (ADmad) [inline tests/TestCase/View/Helper/FlashHelperTest.php]: This change causes tests to fail on 3.5 and we are trying to maintain 3.5 compatibility too. So to make this work on both 3.5 and 3.6 you remove the `use` statement for session class and instead inside `setUp()` use a `class_exists()` check and accordingly use appropriate class to create instance.

**Comment by cewi** [comment]
did you mean like so?

File: tests/TestCase/View/Helper/FlashHelperTest.php [hunk 3]
```diff
@@ -34,8 +33,17 @@ public function setUp()
     {
         parent::setUp();
         $this->View = new View();
-        $session = new Session();
-        $this->View->request = new Request(['session' => $session]);
+
+        // load appropriate session class according to Cake version
+        if (!class_exists('\Cake\Http\Session')) {
```
Comment (ADmad) [inline tests/TestCase/View/Helper/FlashHelperTest.php]: You can remove the negation and swap the `if` and `else` parts :slightly_smiling_face:

Comment (cewi) [reply tests/TestCase/View/Helper/FlashHelperTest.php]: yeah, this complexity is really useless

**Comment by ADmad** [comment]
Thanks!

**Pull request merged**